sscanss/app/dialogs/misc.py

```diff
@@ -1369,3 +1371,325 @@ def plot(self):
         self.axes.set_ylabel('Path Length (mm)')
         self.axes.minorticks_on()
         self.canvas.draw()
+
+
+class CurveEditor(QtWidgets.QDialog):
+    """Creates a UI for modifying a volumes transfer function curve
+
+    :param volume: volume object
+    :type volume: Volume
+    :param parent: main window instance
+    :type parent: MainWindow
+    """
+    def __init__(self, volume, parent):
+        super().__init__(parent)
+
+        self.parent = parent
+
+        self.last_pos = None
+        self._selected_index = 0
+
+        histogram, edge = np.histogram(volume.data, bins=256)
```

Comment:
I suspect this is the slow bit. It might help to move this until after the window has opened.

It might help to pass in the range, as Volume has already calculated a min and max.

Reply:
I am not sure how useful the histogram in the background will be for the user but I can move this into volume init instead